osu.Game/IO/Archives/ZipArchiveReader.cs

```diff
@@ -43,7 +58,9 @@ public override void Dispose()
             archiveStream.Dispose();
         }
 
-        public override IEnumerable<string> Filenames => archive.Entries.Select(e => e.Key).ToArray();
+        private static bool canBeIgnored(IEntry entry) => filename_ignore_list.Any(ignoredName => entry.Key.IndexOf(ignoredName, StringComparison.InvariantCultureIgnoreCase) >= 0);
```

Comment:
Just something to be aware of, but this may match `test/MyMapBackground.DS_Store.jpg`. It's probably never going to be an issue, but could be fixed by matching the end of the string or otherwise.

Reply:
Yeah, I felt like I was skimping on doing the checks properly, as I didn't want to split into two lists (one for full path and one for filename only). I think I'm going to go back and make it safer so that it can't ignore files wrongly.

Reply:
I don't think the path needs to be considered, can just do an `EndsWith` or similar (and optionally a check for the path separator). We be ignoring these files no matter what path they are in.

Reply:
Sure, but the `__MACOSX` folder might require a different way of handling it entirely as it can show in the middle of the path (and I think ZIP folder entries are separate from the entries of files contained within, if not entirely nonexistent), that's why the split into two ignore lists would be necessary. I'd have to actually check to be certain, though.

Reply:
Aaah, I forgot this can be a folder.

To be honest, I think it's fine to just merge as-is. Not worth spending extra effort on.